source/elens.f90

```diff
@@ -383,35 +381,41 @@ end subroutine elens_postInput
 !  - total beam energy [MeV]
 !  - outer radius [mm]
 ! ================================================================================================ !
-real(kind=fPrec) function eLensTheta( len, Int, Ekin, Etot, r2 )
+subroutine eLensTheta( j, Etot )
 
   use floatPrecision
   use mathlib_bouncer
   use numerical_constants
   use physical_constants
   use crcoall
   use mod_common
+  use mod_hions, only : nucm0, zz0
 
   implicit none
 
-  real(kind=fPrec) gamma, beta_e, beta_b, brho, len, Int, Ekin, Etot, r2
+  integer j
+  real(kind=fPrec) gamma, beta_e, beta_b, brho, Etot
 
-  gamma  = Ekin*c1m3/pmae+1 ! from kinetic energy
+  gamma  = elens_Ek(j)*c1m3/pmae+one ! from kinetic energy
   beta_e = sqrt((gamma+one)*(gamma-one))/(gamma)
-  gamma  = Etot/pma ! from total energy
+  gamma  = Etot/nucm0                ! from total energy
   beta_b = sqrt((gamma+one)*(gamma-one))/(gamma)
-  brho   = Etot/(clight*c1m6)
+  brho   = Etot/(clight*c1m6)/zz0
 
-  ! r2: from mm to m
-  ! theta: from rad to mrad
-  eLensTheta = len*abs(Int)/(2*pi*eps0*brho*clight**2*r2*c1m3)*c1e3
-  if(Int < zero) then
-    eLensTheta = eLensTheta*(one/(beta_e*beta_b)+one)
+  ! r2: from mm to m (c1m3)
+  ! theta: from rad to mrad (c1e3)
+  elens_theta_r2(j) = elens_len(j)*abs(elens_I(j))/(two*pi*eps0*brho*clight**2*elens_r2(j)*c1m3)*c1e3
```

Comment:
This line needs brackets for consistent rounding.

Reply:
done - I will anyway review the entire code of elenses to make sure that there are no other left-overs of this type